packages/mdc-linear-progress/_mixins.scss

```diff
@@ -18,28 +18,252 @@
 @import "@material/theme/mixins";
 @import "@material/theme/variables"; // for mdc-theme-prop-value
+@import "./variables";
+@import "./keyframes";
+
+//
+// Public
+//
+
+@mixin mdc-linear-progress-core-styles($query: mdc-feature-all()) {
+  $feat-structure: mdc-feature-create-target($query, structure);
+  $feat-animation: mdc-feature-create-target($query, animation);
+
+  @include mdc-feature-targets($feat-animation) {
+    @include mdc-linear-progress-primary-indeterminate-translate-keyframes_;
+    @include mdc-linear-progress-primary-indeterminate-scale-keyframes_;
+    @include mdc-linear-progress-secondary-indeterminate-translate-keyframes_;
+    @include mdc-linear-progress-secondary-indeterminate-scale-keyframes_;
+    @include mdc-linear-progress-buffering-keyframes_;
+    @include mdc-linear-progress-primary-indeterminate-translate-reverse-keyframes_;
+    @include mdc-linear-progress-secondary-indeterminate-translate-reverse-keyframes_;
+    @include mdc-linear-progress-buffering-reverse-keyframes_;
+  }
+
+  .mdc-linear-progress {
+    @include mdc-feature-targets($feat-structure) {
+      position: relative;
+      width: 100%;
+      height: 4px;
+      transform: translateZ(0);
+      overflow: hidden;
+    }
+
+    @include mdc-feature-targets($feat-animation) {
+      transition: mdc-animation-exit-temporary(opacity, 250ms);
+    }
+
+    &__bar {
+      @include mdc-feature-targets($feat-structure) {
+        position: absolute;
+        width: 100%;
+        height: 100%;
+        animation: none;
+        transform-origin: top left;
+      }
+
+      @include mdc-feature-targets($feat-animation) {
+        transition: mdc-animation-exit-temporary(transform, 250ms);
+      }
+    }
+
+    &__bar-inner {
+      @include mdc-feature-targets($feat-structure) {
+        display: inline-block;
+        position: absolute;
+        width: 100%;
+        height: 100%;
+        animation: none;
+      }
+    }
+
+    &__buffering-dots {
+      @include mdc-feature-targets($feat-structure) {
+        position: absolute;
+        width: 100%;
+        height: 100%;
+        background-repeat: repeat-x;
+        background-size: 10px 4px;
+      }
+
+      @include mdc-feature-targets($feat-animation) {
+        animation: buffering 250ms infinite linear;
+      }
+    }
+
+    &__buffer {
+      @include mdc-feature-targets($feat-structure) {
+        position: absolute;
+        width: 100%;
+        height: 100%;
+        transform-origin: top left;
+      }
+
+      @include mdc-feature-targets($feat-animation) {
+        transition: mdc-animation-exit-temporary(transform, 250ms);
+      }
+    }
+
+    &__primary-bar {
+      @include mdc-feature-targets($feat-structure) {
+        transform: scaleX(0);
+      }
+    }
+
+    &__secondary-bar {
+      @include mdc-feature-targets($feat-structure) {
+        visibility: hidden;
+      }
+    }
+
+    @include mdc-linear-progress-indeterminate_($query: $query);
```

Comment:
I had to break these out because stylelint was complaining about too much nesting.